SofaKernel/modules/SofaSimulationGraph/DAGNode.cpp

```diff
@@ -132,7 +132,8 @@ DAGNode::~DAGNode()
 /// Create, add, then return the new child of this Node
 Node::SPtr DAGNode::createChild(const std::string& nodeName)
 {
-    DAGNode::SPtr newchild = sofa::core::objectmodel::New<DAGNode>(nodeName);
+    msg_warning_when(nodeName.empty(), getName()) << "Empty name given to child: Renaming to \"unnamed\" as this can lead to unexpected behaviors.";
+    DAGNode::SPtr newchild = sofa::core::objectmodel::New<DAGNode>(nodeName.empty()?"unnamed":nodeName);
```

Comment:
it would be nice to have an increment like: unnamed_0001 etc... to avoid component with same name.

Reply:
Few years ago there was discussion about duplicated names in Sofa. 
https://github.com/sofa-framework/sofa/issues/362

The problem was that to be implemented this requires to search for duplicated names...which, because of current implementation, is a linear operation and we didn't wanted to have that at runtime.

Reply:
OK... so this PR's setup (the warning I've set + no action if duplicated name) is best?
Should I leave it as it is?

Reply:
IMO yes. 

Duplicated name is a different topic.

Reply:
and even a random number? without checking the same random number has been used

Reply:
If so I prefer adding a duplicate check. A random number sounds a bit like a hack to me :/

Reply:
So given this morning discussion it was agreed to have an auto-incrementing name restricted to when there is a  replacement with "unnamed" (instead of duplicate detection/removale) at every addChild/createChild.).